core/storage/suggestion/gae_models.py

```diff
@@ -1990,3 +1997,1360 @@ def export_data(
+            topic_ids: List[str]|None.
+                List of topics user has contributions.
+            last_activity: int. Number of days within which a user has
+                contributed.
+            language_code: str. Language Code for filter.
+
+        Returns:
+            3-tuple(sorted_results, next_offset, more). where:
+                sorted_results:
+                    list(TranslationSubmitterTotalContributionStatsModel).
+                    The list of models with supplied filters and sort_by.
+                next_offset: int. Number of results to skip in next batch.
+                more: bool. If True, there are (probably) more results after
+                    this batch. If False, there are no further results
+                    after this batch.
+        """
+        sort = -cls.recent_performance
+        if sort_by == (
+                constants.CD_ADMIN_STATS_SORT_OPTIONS[
+                    'IncreasingLastActivity']):
+            sort = -cls.last_contribution_date
+        elif sort_by == (
+                constants.CD_ADMIN_STATS_SORT_OPTIONS[
+                    'DecreasingLastActivity']):
+            sort = cls.last_contribution_date
+        elif sort_by == (
+                constants.CD_ADMIN_STATS_SORT_OPTIONS[
+                    'IncreasingPerformance']):
+            sort = cls.recent_performance
+        elif sort_by == (
+                constants.CD_ADMIN_STATS_SORT_OPTIONS[
+                    'DecreasingAccuracy']):
+            sort = -cls.overall_accuracy
+        elif sort_by == (
+                constants.CD_ADMIN_STATS_SORT_OPTIONS[
+                    'IncreasingAccuracy']):
+            sort = cls.overall_accuracy
+        elif sort_by == (
+                constants.CD_ADMIN_STATS_SORT_OPTIONS[
+                    'DecreasingSubmissions']):
+            sort = -cls.submitted_translations_count
+        elif sort_by == (
+                constants.CD_ADMIN_STATS_SORT_OPTIONS[
+                    'IncreasingSubmissions']):
+            sort = cls.submitted_translations_count
+
+        # The first sort property must be the same as the property to which
+        # an inequality filter is applied. Thus, the inequality filter on
+        # last_activity can not be used here.
+        if topic_ids is not None:
+            sort_query = cls.query(
+                datastore_services.all_of(
+                    cls.language_code == language_code,
+                    cls.topic_ids_with_translation_submissions.IN(topic_ids)
+                    )).order(sort)
+        else:
+            sort_query = cls.query(
+                datastore_services.all_of(
+                    cls.language_code == language_code
+                    )).order(sort)
+
+        sorted_results: List[
+            TranslationSubmitterTotalContributionStatsModel] = []
+        num_models_per_fetch = 100
+        today = datetime.date.today()
+
+        if last_activity is not None:
```

Comment:
You can't use a filter for this in the datastore query above?

Reply:
We can't because of the property of gae models that: 
**The first sort property must be the same as the property to which an inequality filter is applied.** 
As we are sorting on multiple metrics we have to go by this method to apply filter for last_activity.

Reply:
I think you can make the comment in line 2233 more clear. Be specific about where the restriction comes from, provide a link, give the reader enough context on what you're trying to do and why it can't be done. Assume your reader has no knowledge of GAE query restrictions or this conversation.

It's fine for the comment to be multiple paragraphs long, as long as it's clear to any developer who reads it.

Reply:
I have added a reference link to gae docs stating that error, where a new developer can easily understand why we didn't used the inequality filter with the sort query.